server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -71,6 +71,20 @@ public String getProgramTitle() {
     return programDefinition.localizedName().getOrDefault(applicantData.preferredLocale());
   }
 
+  @Override
+  public ImmutableList<String> getStoredFileKeys() {
+    return getAllActiveBlocks().stream()
+        .filter(Block::isFileUpload)
+        .flatMap(
+            block ->
+                block.getQuestions().stream()
+                    .filter(question -> question.isAnswered() && question.isFileUploadQuestion()))
```

Comment:
all of these can be their own operations in the parent stream.

.flatMap(Block::getQuestions)
.filter(Question::isAnswered)
.filter(Question::isFileUploadQuestion)

Reply:
done. though i was disappointed in java to learn that i couldn't do `.flatMap(Block::getQuestions)` because flatMap doesn't know how to turn an `ImmutableList` into a stream so i had to do `.flatMap(block -> block.getQuestions().stream())`